metadata-ingestion/src/datahub/ingestion/source/powerbi/powerbi.py

```diff
@@ -885,17 +920,17 @@ def validate_dataset_type_mapping(self):
             if key not in powerbi_data_platforms:
                 raise ValueError(f"PowerBI DataPlatform {key} is not supported")
 
-    def get_workunits(self) -> Iterable[MetadataWorkUnit]:
+    def get_workunits_internal(self) -> Iterable[MetadataWorkUnit]:
         """
         Datahub Ingestion framework invoke this method
         """
         logger.info("PowerBi plugin execution is started")
         # Validate dataset type mapping
         self.validate_dataset_type_mapping()
         # Fetch PowerBi workspace for given workspace identifier
-        for workspace_id in self.get_workspace_ids():
-            logger.info(f"Scanning workspace id: {workspace_id}")
-            workspace = self.powerbi_client.get_workspace(workspace_id, self.reporter)
+        for workspace in self.get_allowed_workspaces():
+            logger.info(f"Scanning workspace id: {workspace.id}")
+            self.powerbi_client.fill_workspace(workspace, self.reporter)
```

Comment:
If an exception happens, we fail the whole ingestion, is there a case we want to continue the ingestion even though there was an error parsing a workspace?  (like the user did not have permission to a dashboard, and it causes permission error)

Reply:
permission errors are handled